WhateverGreen/kern_igfx.cpp

```diff
@@ -248,3 +349,493 @@ bool IGFX::wrapAcceleratorStart(IOService *that, IOService *provider) {
+					platformInformationList[i].fBTTableOffsetIndexHDMI = callbackIGFX->framebufferPatch.fBTTableOffsetIndexHDMI;
+				
+				if (callbackIGFX->framebufferPatchFlags.bits.FPFCameliaVersion)
+					platformInformationList[i].cameliaVersion = callbackIGFX->framebufferPatch.cameliaVersion;
+				
+				if (callbackIGFX->framebufferPatchFlags.bits.FPFNumTransactionsThreshold)
+					platformInformationList[i].fNumTransactionsThreshold = callbackIGFX->framebufferPatch.fNumTransactionsThreshold;
+				
+				if (callbackIGFX->framebufferPatchFlags.bits.FPFVideoTurboFreq)
+					platformInformationList[i].fVideoTurboFreq = callbackIGFX->framebufferPatch.fVideoTurboFreq;
+				
+				if (callbackIGFX->framebufferPatchFlags.bits.FPFBTTArraySliceAddr)
+					platformInformationList[i].fBTTArraySliceAddr = callbackIGFX->framebufferPatch.fBTTArraySliceAddr;
+				
+				if (callbackIGFX->framebufferPatchFlags.bits.FPFBTTArrayNormalAddr)
+					platformInformationList[i].fBTTArrayNormalAddr = callbackIGFX->framebufferPatch.fBTTArrayNormalAddr;
+				
+				if (callbackIGFX->framebufferPatchFlags.bits.FPFBTTArrayHDMIAddr)
+					platformInformationList[i].fBTTArrayHDMIAddr = callbackIGFX->framebufferPatch.fBTTArrayHDMIAddr;
+				
+				if (callbackIGFX->framebufferPatchFlags.bits.FPFSliceCount)
+					platformInformationList[i].fSliceCount = callbackIGFX->framebufferPatch.fSliceCount;
+				
+				if (callbackIGFX->framebufferPatchFlags.bits.FPFEuCount)
+					platformInformationList[i].fEuCount = callbackIGFX->framebufferPatch.fEuCount;
+			}
+		}
+	}
+	
+	for (int i=0; i<MAX_FRAMEBUFFER_PATCH_COUNT; i++) {
+		if (callbackIGFX->framebufferPatches[i].find == nullptr || callbackIGFX->framebufferPatches[i].replace == nullptr)
+			continue;
+		
+		if (callbackIGFX->framebufferPatches[i].framebufferId != framebufferId)
+		{
+			framebufferAddress = callbackIGFX->findFramebufferId(callbackIGFX->framebufferPatches[i].framebufferId, (uint8_t *)callbackIGFX->gPlatformInformationList, framebufferSize);
+			framebufferId = callbackIGFX->framebufferPatches[i].framebufferId;
+		}
+		
+		KernelPatcher::LookupPatch patch;
+		
+		patch.kext = callbackIGFX->currentFramebuffer;
+		patch.find = (const uint8_t *)callbackIGFX->framebufferPatches[i].find->getBytesNoCopy();
+		patch.replace = (const uint8_t *)callbackIGFX->framebufferPatches[i].replace->getBytesNoCopy();
+		patch.size = min(callbackIGFX->framebufferPatches[i].find->getLength(), callbackIGFX->framebufferPatches[i].replace->getLength());
+		patch.count = callbackIGFX->framebufferPatches[i].count;
+		
+		callbackIGFX->patcher->applyLookupPatch(&patch, framebufferAddress, framebufferSize + 0x100);
+		
+		callbackIGFX->framebufferPatches[i].find->release();
+		callbackIGFX->framebufferPatches[i].replace->release();
+	}
+	
+	return r;
+}
+
+uint8_t *IGFX::findFramebufferId(uint32_t framebufferId, uint8_t *startingAddress, size_t maxSize) {
+	uint8_t *startAddress = startingAddress;
+	uint8_t *endAddress = startingAddress + maxSize;
```

Comment:
This should be `uint8_t *endAddress = startingAddress + maxSize - sizeof(uint32_t);` or you risk addressing invalid memory. While unlikely, it is best to stay safe.